mtags/src/main/scala/scala/meta/internal/mtags/ScalaToplevelMtags.scala

```diff
@@ -106,14 +104,23 @@ class ScalaToplevelMtags(
       )
 
     def needEmitFileOwner(region: Region): Boolean =
-      !sourceTopLevelAdded && region.produceSourceToplevel
+      region.produceSourceToplevel
     def needToParseBody(expect: ExpectTemplate): Boolean =
       (includeInnerClasses || expect.isPackageBody) && !expect.ignoreBody
     def needToParseExtension(expect: ExpectTemplate): Boolean =
       includeInnerClasses && expect.isExtension && !expect.ignoreBody
+
     def nextIsNL: Boolean = {
       scanner.nextToken()
-      isNewline
+      scanner.curr.token match {
```

Comment:
`acceptTrivia` would only consume whitespace and comments, not tell me if there was a newline.
Also, I think it may mess up counting indent in the following line

Reply:
Yeah, sorry, my bad. I didn't realised the token for newline in `WHITESPACE` and that `acceptTrivia` would consume it.